File: 822A - I'm bored with life.java
//https://codeforces.com/problemset/problem/822/A
//822A - I'm bored with life


import java.util.Scanner;
import java.math.BigInteger;

public class Main{
    static BigInteger fact(int N){
        BigInteger b = new BigInteger("1");
        for(int i=2;i<=N;i++){
            b = b.multiply(BigInteger.valueOf(i));
        }
        return b;
    }
    public static void main(String args[]){
        Scanner sc = new Scanner(System.in);
        int a = sc.nextInt();
        int b = sc.nextInt();
        BigInteger b1 = fact(a); 
        BigInteger b2 = fact(b);
        BigInteger result;
        result = b1.gcd(b2);
        System.out.println(result);
    }
}

File: 822A - I'm bored with life.cpp
//https://codeforces.com/problemset/problem/822/A
//822A - I'm bored with life
#include <bits/stdc++.h> 
using namespace std; 
int factorial(int n) { 
    return (n == 1 || n == 0) ? 1 : factorial(n - 1) * n; 
}  
int gcdOfFactorial(int arr[], int n) { 
    int minm = arr[0]; 
    for (int i = 1; i < n; i++) 
        minm = minm > arr[i] ? arr[i] : minm; 
    return factorial(minm); 
} 
  
int main() { 
    int arr[2];
    cin>>arr[0]>>arr[1]; 
    cout << gcdOfFactorial(arr, 2); 
    return 0; 
} 